orchagent/routeorch.cpp

```diff
@@ -490,14 +490,20 @@ bool RouteOrch::removeNextHopGroup(IpAddresses ipAddresses)
 
         m_syncdNextHopGroups.erase(ipAddresses);
     }
-
     return true;
 }
 
 void RouteOrch::addTempRoute(IpPrefix ipPrefix, IpAddresses nextHops)
 {
     SWSS_LOG_ENTER();
 
+    /* Check if the same temporary route is already inserted or not */
+    if (m_syncdTempRoutes.find(ipPrefix) != m_syncdTempRoutes.end() &&
+        m_syncdTempRoutes[ipPrefix] == nextHops)
+    {
+        return;
+    }
+
```

Comment:
I would put this logic outside the addTempRoute, because it determines whether we should add temproute or not.